### PR TITLE
docs(template): document the reactive context

### PR DESCRIPTION
**Pull request by BioPhoton**

### CODE REVIEW

File: libs/template/docs/reactive-context.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# The Reactive Context 
+
+If we think about any process, e. g. a http request, we can differentiate different states in it. 
+The request can start, result in a result or throws an error. After that the process is completed.
+
+This fits to pretty much any process, thy just differ in the number of emissions.
```
Comment (kajetansw) [inline libs/template/docs/reactive-context.md:6]:
```suggestion
This fits pretty much any process, they just differ in the number of emissions.
```
I think some typo got in ;) Is this the way it should be?

File: libs/template/docs/reactive-context.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+- `next`
+- `error`
+- `complete` 
+
+Where `next` can have zero to infinite emissions, and `error` and `complete` zero to 1 emissions.
+
+This differentiation works perfect in code, however when we start to project those channels and their changes into HTML
+we sometimes end up implementing a hack for another potential state that is not provided by Angular.
+ 
+In the above example with the http request, we may want to indicates that the process was started,
```
Comment (kajetansw) [inline libs/template/docs/reactive-context.md:18]:
```suggestion
In the above example with the http request, we may want to indicate, that the process was started, 
```

File: libs/template/docs/reactive-context.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+Where `next` can have zero to infinite emissions, and `error` and `complete` zero to 1 emissions.
+
+This differentiation works perfect in code, however when we start to project those channels and their changes into HTML
+we sometimes end up implementing a hack for another potential state that is not provided by Angular.
+ 
+In the above example with the http request, we may want to indicates that the process was started, 
+but no result, neither a value nor an error or complete arrived yet from the process.
+
+This leads to not only clutter HTML, but more worse to `eager template`.
```
Comment (kajetansw) [inline libs/template/docs/reactive-context.md:21]:
```suggestion
This leads not only to a cluttered HTML, but, what's worse, to the `eager template`.
```
"More worse" is just "worse" ;)

File: libs/template/docs/reactive-context.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+```typescript
+import { Notification } from 'rxjs';
+export type RxNotificationKind = 'rxSuspense' | 'rxNext' | 'rxError' | 'rxComplete';
+type NotificationExtract = 'value' | 'hasValue' | 'error';
+export type RxNotification<T> = Pick<Notification<T>, NotificationExtract> & { kind: RxNotificationKind };
+```
+
+---
+
+As a sum up we now know that `rx-angular/template` provides an extended reactive context with the `suspense` channel.
```
Comment (kajetansw) [inline libs/template/docs/reactive-context.md:103]:
```suggestion
As a sum up we now know that `@rx-angular/template` provides an extended reactive context with the `suspense` channel.
```

File: libs/template/docs/reactive-context.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
+# The Reactive Context 
+
+If we think about any process, e. g. a http request, we can differentiate different states in it. 
+The request can start, result in a result or throws an error. After that the process is completed.
```
Comment (BioPhoton) [inline libs/template/docs/reactive-context.md:4]:
```suggestion
The request can start, result in a valid response or throws an error. After that the process is completed.
```

File: libs/template/docs/reactive-context.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
+# The Reactive Context 
+
+If we think about any process, e. g. a http request, we can differentiate different states in it. 
+The request can start, result in a valid response or throws an error. After that the process is completed.
+
+This fits pretty much any process, they just differ in the number of emissions.
```
Comment (BioPhoton) [inline libs/template/docs/reactive-context.md:6]:
```suggestion
These states fit pretty much any process, they just differ in the number of emissions.
```

File: libs/template/docs/reactive-context.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
+# The Reactive Context
```
Comment (BioPhoton) [inline libs/template/docs/reactive-context.md:1]:
```suggestion
# The extended reactive context in RxAngular
```

**Pull request merged**